docs/feature-requests/005-editable-diff.md

```diff
@@ -0,0 +1,175 @@
+# Editable Diff
+
+## :memo: Summary
+
+Inline editing within a diff view (i.e. `MultiFilePatchView`).
+
+## :checkered_flag: Motivation
+
+This can save users the trouble of needing to toggle to an editor and back again when they notice typos, `console.log()` statements, or `.only()` in tests when reviewing changes right before committing.
+
+This also serves as a building block of the bigger [PR review workflow](https://github.com/atom/github/blob/master/docs/feature-requests/003-pull-request-review.md) we want to eventually implement.
+
+## 🤯 Explanation
+
+### What is editable?
+
+Currently, we use diff view in several places; which ones of them are editable?
+  - Unstaged Changes: *editable*
+  - Staged Changes: *editable* but with [unresolved questions](#question-unresolved-questions)
+  - All staged changes (aka Commit Preview): *editable* but with [unresolved questions](#question-unresolved-questions)
+  - Commit Detail Item: *NOT editable*
+  - Changed File Tab in PR: *editable **only** if it's a checked out PR*, also with [unresolved questions](#question-unresolved-questions)
+
+### Workflow of editing a diff
+
+#### 1. The entry point of editable state
+
+- By default, a diff view is editable (with some [exceptions](#what-is-editable) listed above). By clicking into any diff, a blinking caret will appear to signify users they can start editing.
+  ![editable](https://user-images.githubusercontent.com/378023/50758847-b27e3f00-12a6-11e9-8c2e-6c0bf085fdbd.gif)
+- If a diff is read-only, there will be no caret. The mouse cursor shows as "default" arrow. Selecting still works.
+  ![non-editable](https://user-images.githubusercontent.com/378023/50758848-b27e3f00-12a6-11e9-996c-8a458196e070.gif)
+
+
+#### 2. Within an editable state
+- The editable state behaves almost like a normal editor, except that the green background of added lines and red background of deleted lines will be kept.
+- Behaviours such as stage/unstage hunk and selected line will remain the same, since the line that is being actively edited will still be considered as "selected", hence show up as highlighted.
+- A deleted line will still be visible within the editable state, but it _is not editable_. That will be communicated to users in a very clear visual manner:
+  - The cursor for hovering over deleted lines will show up as an arrow as opposed to a text cursor.
+  - Deleted lines can still be selected (via mouse or keyboard) but no caret will be visible; basically same as current behaviour.
+  ![deleted](https://user-images.githubusercontent.com/378023/50759911-b3fd3680-12a9-11e9-8332-abf211e0743c.gif)
+- As a user makes edits in a diff view, the file name will have some visual cue indicating there are unsaved modifications, akin to when a regular atom editor has unsaved changes.
```

Comment:
Might be useful to also show the modified indicator (blue dot) on the tab? Just like for regular files. Then it's easier to see if a diff has unsaved edits when it's not the active tab or the file header is "scrolled out of view".

I _might_ be even enough to **only** have the indicator on the tabs, even for multi-file diffs.